prefect_census/syncs.py

```diff
@@ -207,3 +219,170 @@ def my_flow():
             f"Triggered sync run with ID: {run_id} ended with unexpected"
             f"status {final_run_status}"
         )
+
+
+class CensusSync(JobBlock):
+    """A Job Block for triggering a Census sync run and waiting for completion.
+
+    Attributes:
+        credentials: Credentials for authenticating with Census.
+        sync_id: The ID of the sync to trigger.
+        force_full_sync: If `True`, a full sync will be triggered.
+        max_wait_seconds: Maximum number of seconds to wait for sync to complete.
+        poll_frequency_seconds: Number of seconds to wait in between checks
+            for run completion.
+
+    Example:
+        ```python
+        from prefect import flow
+        from prefect_census import CensusSync, run_census_sync
+
+        @flow
+        def my_census_flow():
+            census_sync = CensusSync.load("BLOCK_NAME")
+
+            # do some setup
+
+            run_census_sync(census_sync)
+
+            # do some cleanup
+        ```
+    """
+
+    credentials: CensusCredentials = Field(
+        default=...,
+        description="Credentials for authenticating with Census.",
+    )
+
+    sync_id: int = Field(
+        default=...,
+        description="The ID of the sync to trigger.",
+    )
+
+    force_full_sync: bool = Field(
+        default=False,
+        description="If `True`, a full sync will be triggered.",
+    )
+
+    max_wait_seconds: int = Field(
+        default=3600,
+        description="The maximum number of seconds to wait for the sync to complete.",
+    )
+
+    poll_frequency_seconds: int = Field(
+        default=5,
+        description="Number of seconds to wait between sync status checks.",
+    )
+
+    _block_type_name = "Census Sync"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/3oznRx2UFkd2XyqNkEZpzB/4e0967a828aec5e2527cedadf8d24e8a/llmjpn8a0pgu8szjmnyi.webp?h=250"  # noqa
+
```

Comment:
This is a short tagline just for the UI.
```suggestion
_description="Runs a Census sync"
```